app/code/Magento/Contact/Controller/Index/Post.php

```diff
@@ -101,4 +63,57 @@ private function getDataPersistor()
 
         return $this->dataPersistor;
     }
+
+    /**
+     * @param array $post Post data from contact form
+     */
+    protected function sendEmail($post)
+    {
+        $storeScope = \Magento\Store\Model\ScopeInterface::SCOPE_STORE;
+        $transport = $this->_transportBuilder
+            ->setTemplateIdentifier($this->scopeConfig->getValue(self::XML_PATH_EMAIL_TEMPLATE, $storeScope))
+            ->setTemplateOptions(
+                [
+                    'area' => \Magento\Backend\App\Area\FrontNameResolver::AREA_CODE,
+                    'store' => \Magento\Store\Model\Store::DEFAULT_STORE_ID,
+                ]
+            )
+            ->setTemplateVars(['data' => new \Magento\Framework\DataObject($post)])
+            ->setFrom($this->scopeConfig->getValue(self::XML_PATH_EMAIL_SENDER, $storeScope))
+            ->addTo($this->scopeConfig->getValue(self::XML_PATH_EMAIL_RECIPIENT, $storeScope))
+            ->setReplyTo($post['email'])
+            ->getTransport();
+
+        $transport->sendMessage();
+    }
+
+    private function isPostRequest()
+    {
+        /** @var Request $request */
+        $request = $this->getRequest();
+        return !empty($request->getPostValue());
+    }
+
+    /**
+     * @return array
+     * @throws \Exception
+     */
+    protected function validatedParams()
+    {
+        $request = $this->getRequest();
+        if (trim($request->getParam('name')) === '') {
+            throw new LocalizedException(__('Name is missing'));
+        }
+        if (trim($request->getParam('comment')) === '') {
+            throw new LocalizedException(__('Comment is missing'));
+        }
+        if (false === \strpos($request->getParam('email'), '@')) {
```

Comment:
I did that on purpose because I don't believe in email address validation via regular expressions (relevant: https://hackernoon.com/the-100-correct-way-to-validate-email-addresses-7c4818f24643). If there's a decision to use the MX check, the ZF2 email validator would be useful again.